nengo/builder/connection.py

```diff
@@ -190,9 +190,6 @@ def get_prepost_signal(is_pre):
         gain = model.params[conn.post_obj.ensemble].gain[post_slice]
         weights = multiply(gain, weights)
 
-    if conn.learning_rule is not None and weights.ndim < 2:
-        raise ValueError("Learning connection must have full transform matrix")
```

Comment:
Is it needed for the other learning rules considering the checks done to validate the LearningRuleType param?